packages/admin/admin/src/mui/menu/CollapsibleItem.tsx

```diff
@@ -1,96 +1,192 @@
 
+import { MenuCollapsibleItemClassKey, styles } from "./CollapsibleItem.styles";
+import { MenuContext } from "./Context";
 import { MenuItem, MenuItemProps } from "./Item";
 import { MenuItemRouterLinkProps } from "./ItemRouterLink";
 
-export type MenuCollapsibleItemClassKey = "root" | "childSelected" | "listItem" | "open";
-
-const styles = (theme: Theme) =>
-    createStyles<MenuCollapsibleItemClassKey, MenuCollapsibleItemProps>({
-        root: {},
-        childSelected: {
-            color: theme.palette.primary.main,
-            "& $listItem": {
-                "& [class*='MuiListItemText-root']": {
-                    color: theme.palette.primary.main,
-                },
-                "& [class*='MuiListItemIcon-root']": {
-                    color: theme.palette.primary.main,
-                },
-            },
-        },
-        listItem: {},
-        open: {},
-    });
-
 export interface MenuLevel {
     level?: 1 | 2 | 3;
 }
 
-type MenuChild = React.ReactElement<MenuItemRouterLinkProps>;
+export type MenuChild = React.ReactElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>;
 
 export interface MenuCollapsibleItemProps extends MenuItemProps {
     children: MenuChild | MenuChild[];
     openByDefault?: boolean;
-    openedIcon?: React.ReactNode;
-    closedIcon?: React.ReactNode;
+    openedIcon?: React.JSXElementConstructor<SvgIconProps>;
+    closedIcon?: React.JSXElementConstructor<SvgIconProps>;
+    isMenuOpen?: boolean;
 }
 
 const CollapsibleItem: React.FC<WithStyles<typeof styles> & MenuCollapsibleItemProps> = ({
     classes,
     level,
     primary,
     secondary,
+    isMenuOpen,
     icon,
     openByDefault = false,
-    openedIcon = <ChevronUp />,
-    closedIcon = <ChevronDown />,
+    openedIcon: OpenedIcon,
+    closedIcon: ClosedIcon,
     children,
     ...otherProps
 }) => {
-    const itemLevel: MenuLevel["level"] = level ? level : 1;
-    let hasSelectedChild = false;
+    OpenedIcon = OpenedIcon || (isMenuOpen ? ChevronUp : ChevronRight);
+    ClosedIcon = ClosedIcon || (isMenuOpen ? ChevronDown : ChevronRight);
+
+    const { drawerVariant } = React.useContext(MenuContext);
+    const itemLevel: MenuLevel["level"] = level ?? 1;
+    const hasSelectedChild = useRef(false);
     const location = useLocation();
 
-    const childElements = React.Children.map(children, (child: MenuChild) => {
-        if (matchPath(location.pathname, { path: child.props.to, strict: true })) {
-            hasSelectedChild = true;
-        }
+    const [isSubmenuOpen, setIsSubmenuOpen] = React.useState<boolean>(openByDefault || hasSelectedChild.current);
+    const [anchorEl, setAnchorEl] = React.useState<HTMLElement | undefined>(undefined);
 
-        const newItemLevel = itemLevel + 1;
+    React.useEffect(() => {
+        // set open state manually to false to avoid a menu opening when isMenuOpen state changes
+        if (!isMenuOpen) setIsSubmenuOpen(false);
+    }, [isMenuOpen]);
```

Comment:
This should also be possible by doing

```tsx
// set open state manually to false to avoid a menu opening when isMenuOpen state changes
if (!isMenuOpen) {
  setIsSubmenuOpen(false);
}
```

without an effect. Docs: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

Reply:
Just tested this again. Seems like it's needed. Otherwise there is an infinite render loop 😅